services/s3/src/main/java/software/amazon/awssdk/services/s3/S3CrtAsyncClientBuilder.java

```diff
@@ -114,12 +114,30 @@ default S3CrtAsyncClientBuilder credentialsProvider(IdentityProvider<? extends A
      */
     S3CrtAsyncClientBuilder minimumPartSizeInBytes(Long uploadPartSize);
 
+    /**
+     * The amount of memory that CRT is allowed to use when making requests to S3.
+     * <p>
+     * If not provided, the CRT attempts to limit memory usage in an optimal way, based on a number of parameters
+     * such as target throughput. Therefore, only configure the memory limit explicitly when needed.
+     * <p>
+     * Supported range:
+     * <ul>
+     *     <li><b>Min: </b>1 GB</li>
+     *     <li><b>Max: </b>The lowest value of the supplied value and the SIZE_MAX of the system</li>
+     * </ul>
+     *
+     * @param memoryLimitInBytes the memory limit in bytes
+     * @return this builder for method chaining.
+     * @see #targetThroughputInGbps(Double)
+     */
+    S3CrtAsyncClientBuilder memoryLimitInBytes(Long memoryLimitInBytes);
```

Comment:
Suggestion `The amount of native memory that ...`. How about `maxNativeMemoryUsageInBytes` or `maxMemoryUsageInBytes`?